File: checkout-ruby/lib/cart_value_discount.rb
require_relative "./promotional_rule"
class CartValueDiscount < PromotionalRule

  def initialize(type, price)
    @value_type = type
    @total_price = price
  end

  def get_discount_entity_type
    RuleDiscountEntityType::CART
  end

  def get_rule_discount_entity
    total_price
  end

  private
  attr_accessor :total_price
end

File: checkout-java/src/main/java/com/vinterior/checkout/models/CartValueDiscount.java
package com.vinterior.checkout.models;

import com.vinterior.checkout.RuleDiscountValue;
import com.vinterior.checkout.enums.RuleDiscountEntityType;

public class CartValueDiscount extends PromotionalRule {

    private double totalPrice;

    public CartValueDiscount(RuleDiscountValue type, double price) {
        super(type);
        totalPrice = price;
    }

    @Override
    public RuleDiscountEntityType getDiscountEntityType() {
        return RuleDiscountEntityType.CART;
    }

    @Override
    public double getRuleDiscountEntity() {
        return totalPrice;
    }


}
